pkg/tests/end_to_end_tests/generate_jaeger_response_test.go

```diff
@@ -0,0 +1,102 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+//go:build gen_response
+// +build gen_response
+
+// Below line generates the Jaeger API responses.
+//go:generate go test . -run TestGenerateJaegerAPIResponses
+
+package end_to_end_tests
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/golang/snappy"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	"google.golang.org/grpc"
+
+	jaegerproto "github.com/jaegertracing/jaeger/proto-gen/api_v2"
+	jaegertranslator "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+
+	"github.com/stretchr/testify/require"
+	"github.com/timescale/promscale/pkg/internal/testhelpers"
+)
+
+// TestGenerateJaegerAPIResponses is not an actual test, rather a function to spin up jaeger container
+// and store the responses based on the test cases, so that they can be statically compared in the
+// order of the testcases.
+func TestGenerateJaegerAPIResponses(t *testing.T) {
+	jaeger, err := testhelpers.StartJaegerContainer(true)
+	require.NoError(t, err)
+
+	sampleTraces := generateTestTrace()
+	err = insertDataIntoJaeger(fmt.Sprintf("localhost:%s", jaeger.GrpcReceivingPort.Port()), copyTraces(sampleTraces))
+	require.NoError(t, err)
+
+	var store traceResponsesStore
+	client := httpClient{url: "http://localhost:" + jaeger.UIPort.Port()}
+	store.Responses = make([]traceResponse, 0)
+
+	queries := traceQueryCases
+	for i := range queries {
+		tc := queries[i]
+		services := getServices(t, client)
+		operations := getOperations(t, client, tc.service)
+		trace := getTrace(t, client, tc.traceID)
+		traces := getTraces(t, client, tc.service, tc.start, tc.end)
+		store.Responses = append(store.Responses, traceResponse{
+			Services:   services,
+			Operations: operations,
+			Trace:      trace,
+			Traces:     traces,
+		})
+	}
+
+	buffer := new(bytes.Buffer)
+
+	enc := gob.NewEncoder(buffer)
+	require.NoError(t, enc.Encode(&store))
+
+	compressed := make([]byte, 0)
+	compressed = snappy.Encode(compressed, buffer.Bytes())
+
+	// #nosec
+	f, err := os.OpenFile(jaegerQueryResponsesPath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0777)
+	require.NoError(t, err)
+	defer f.Close()
+
+	require.NoError(t, f.Truncate(0))
+
+	_, err = f.Write(compressed)
+	require.NoError(t, err)
```

Comment:
this can be greatly simplified by using a snappy buffered writer. Example:

```
enc := gob.NewEncoder(snappy.NewBufferedWriter(file))
enc.Encode(&store)
```